### PR TITLE
Allow the user to choose the number of bombs to place on the board

### PR DESCRIPTION
**Pull request by lottetreg**
- Add `GameSetup` which is responsible for updating the `GameState` with the number of bombs to place on the board
- Add `Game.start/1` as the new point of entry, which calls `Game.play/1` if `GameSetup.setup/1` didn't receive an exit command
- Pass the number of bombs from the `GameState` into `BombPlacer.place_bombs/2` (`BombPlacer` was already capable of accepting a number of bombs as a second arg)

Refactorings:
- Both `AdjacentBombCount` and `FloodFiller` need to be able to access a tile's adjacent tiles, so extract the common functionality into `AdjacentTileLocations`. Also rework the logic in there to be much more explicit with cardinal and ordinal function names. (Apparently ["ordinal"](https://www.geolounge.com/cardinal-directions-ordinal-directions/) is the term for the "in-between" directions, like North-West, which I didn't know!)
- Since `GameSetup` now also needs to check for the exit command, move this logic into `InputFilter` and use this module inside `Game` and `GameSetup`
- Move all of the "\n"s into a single place, in `Message.format/1` (from [this](https://github.com/lottetreg/minesweeper/pull/14#discussion_r296430852) comment)
- Use function capturing in all the places I wasn't before (from [this](https://github.com/lottetreg/minesweeper/pull/14#discussion_r296432018) comment)
- Remove test for `Board.select_tile/2`, since it doesn't adding anything (from [this](https://github.com/lottetreg/minesweeper/pull/16#discussion_r296720865) comment)

### CODE REVIEW

**Review by ecmendenhall — APPROVED** [review]
Just one comment on inlining a function call. :rocket:

**Comment by lottetreg** [comment]
> Just one comment on inlining a function call. 🚀

Done! Thank you, @ecmendenhall!

File: lib/board.ex [hunk 1]
```diff
@@ -21,9 +21,13 @@ defmodule Board do
   end
 
   def get_tile(board, {row_index, col_index}) do
-    board
-    |> Enum.at(row_index)
-    |> Enum.at(col_index)
+    if out_of_bounds?(board, {row_index, col_index}) do
+      nil
+    else
```
Comment (parkerpersaud) [inline lib/board.ex:26]: I'm more familiar with Java & Scala but seeing a function return `nil` gives me a bit of pause. Are callers of this function handling the `nil` case?

**Review by parkerpersaud — COMMENTED** [review]
Had a question about returning `nil` but otherwise LGTM 🚢 🇮🇹

**Review by parkerpersaud — APPROVED** [review]

File: lib/board.ex (hunk 1 above)
Comment (lottetreg) [reply lib/board.ex:26]: @klkelley thank you! Only one caller of this function is handling the `nil` case, so it makes a lot of sense that this is iffy. I've updated it [here](https://github.com/lottetreg/minesweeper/commit/cdd7f5f798bc79fbf1e4c9c807fc313b78040d41). I should probably change `Board.get_tile/2` to account for cases in which the tile doesn't exist, and then update the callers to handle those cases, right?

Comment (parkerpersaud) [reply lib/board.ex:26]: Nice, yeah I'd still account for the case the tile doesn't exist

**Pull request merged**